# Introduction
After collecting the data, I formed a table with the given dataset and also added another column for cost per shoe. Using the formed table, I created various pivot tables to start analyzing the different given data. 

# Financial Analysis

The average order value(AOV) is indeed \\$3,145.13. However, after deeper analysis by creating a table for this large dataset and looking through the values of each transaction. I have discovered that user_id 607 has a total of 17 transactions valued at \\$704,000 (\\$352 per shoe) each through credit card at 4:00 AM every transaction. In other words, user_id 607 has brought up the overall AOV from \\$754 (excluding user_id 607’s transaction) to \\$3,145.13 (including user_id 607’s transaction. This user has bought a total of 34,000 shoes in the past 30 days valuing almost \\$12 million through a credit card. The series of irregular activity is highly suspicious, and should be overwatched carefully and brought to the data security team to ensure that there is no credit card fraud. Not to mention, the high order amount for shop_id 78 (\\$2,263,800) also skews the data. 

These shops are selling sneakers and they are relatively cheap; however, the unit price is not the only factor that contributes to the movement of AOV. Shopify should also take the quantity sold into consideration. For example, the unit price of shop_id 78’s shoe is \\$25,725, higher than the market average, \\$152 (excluding shop_id 78’s shoe). However, it doesn’t play as much of an impact as user_id 607’s transaction because user_id 607’s buys a total of 2,000 shoes per transaction. This makes the value of each transaction higher than it would be if it was purchased in a smaller quantity. Therefore, if Shopify is trying to evaluate the AOV, they should also take the number of shoes being purchased into consideration.

In [21]:
import pandas as pd
import numpy as np

In [22]:
#Raw data visualization
df = pd.read_csv("RawData.csv")
pd.set_option("display.max_columns", 10)
pd.set_option('display.width', 1000)
print(df)

      order_id  shop_id  user_id  order_amount  total_items payment_method           created_at
0            1       53      746           224            2           cash  2017-03-13 12:36:56
1            2       92      925            90            1           cash  2017-03-03 17:38:52
2            3       44      861           144            1           cash   2017-03-14 4:23:56
3            4       18      935           156            1    credit_card  2017-03-26 12:43:37
4            5       18      883           156            1    credit_card   2017-03-01 4:35:11
...        ...      ...      ...           ...          ...            ...                  ...
4995      4996       73      993           330            2          debit  2017-03-30 13:47:17
4996      4997       48      789           234            2           cash  2017-03-16 20:36:16
4997      4998       56      867           351            3           cash   2017-03-19 5:42:42
4998      4999       60      825        

In [23]:
#Confirming the value of AOV
average = df["order_amount"].mean()
print("AOV: " + "${:,.2f}".format(average))

AOV: $3,145.13


In [4]:
#Sorting the data by order_amount
x = df[df.columns[1:7]]
sort = x.sort_values("order_amount", ascending = False)
print(sort.head(25))

      shop_id  user_id  order_amount  total_items payment_method           created_at
2153       42      607        704000         2000    credit_card   2017-03-12 4:00:00
3332       42      607        704000         2000    credit_card   2017-03-24 4:00:00
520        42      607        704000         2000    credit_card   2017-03-02 4:00:00
1602       42      607        704000         2000    credit_card   2017-03-17 4:00:00
60         42      607        704000         2000    credit_card   2017-03-04 4:00:00
2835       42      607        704000         2000    credit_card   2017-03-28 4:00:00
4646       42      607        704000         2000    credit_card   2017-03-02 4:00:00
2297       42      607        704000         2000    credit_card   2017-03-07 4:00:00
1436       42      607        704000         2000    credit_card   2017-03-11 4:00:00
4882       42      607        704000         2000    credit_card   2017-03-25 4:00:00
4056       42      607        704000         2000    c

In [5]:
#Digging deeper into each user
pivot1 = df.pivot_table(index=(["user_id"]), values=["order_amount"], aggfunc={"order_amount":np.sum, "user_id":np.size})
pivot1 = pivot1.rename(columns={"user_id":"count"})
data = pivot1.sort_values("order_amount", ascending = False)
print(data.head(10))

         order_amount  count
user_id                     
607          11968000     17
878            156936     11
834            108342     18
787             85707     25
969             84269     23
818             82485     19
775             81114     16
915             80992     14
817             80988     16
855             80915     17


In [6]:
#Digging deeper into each users
#pivot1 = df.pivot_table(index=(["user_id"]), values=["order_amount"], aggfunc=np.sum)
#data = pivot1.sort_values("order_amount", ascending = False)
#print(data.head(10))


In [7]:
#Deep into 607
x = df[df.columns[1:7]]
irregular = x[df["user_id"]==607]
print(irregular)

      shop_id  user_id  order_amount  total_items payment_method          created_at
15         42      607        704000         2000    credit_card  2017-03-07 4:00:00
60         42      607        704000         2000    credit_card  2017-03-04 4:00:00
520        42      607        704000         2000    credit_card  2017-03-02 4:00:00
1104       42      607        704000         2000    credit_card  2017-03-24 4:00:00
1362       42      607        704000         2000    credit_card  2017-03-15 4:00:00
1436       42      607        704000         2000    credit_card  2017-03-11 4:00:00
1562       42      607        704000         2000    credit_card  2017-03-19 4:00:00
1602       42      607        704000         2000    credit_card  2017-03-17 4:00:00
2153       42      607        704000         2000    credit_card  2017-03-12 4:00:00
2297       42      607        704000         2000    credit_card  2017-03-07 4:00:00
2835       42      607        704000         2000    credit_card 

In [15]:
#AOV comparison with and without user 607
WO607 = df[df["user_id"]!=607]
avgWO607 = WO607["order_amount"].mean()
print("AOV with 607: " + "${:,.2f}".format(average))
print("AOV without 607: " + "${:,.2f}".format(avgWO607))

AOV with 607: $3,145.13
AOV without 607: $754.09


In [16]:
#Finding the average price of each shoe 
pivot2 = WO607.pivot_table(index=(["shop_id"]), values = ["order_amount", "total_items"], aggfunc=np.sum)
pivot2["avg_price"] =  pivot2["order_amount"]/pivot2["total_items"]
pivot2 = pivot2.sort_values("avg_price", ascending = False)
print(pivot2.head(20))

         order_amount  total_items  avg_price
shop_id                                      
78            2263800           88    25725.0
42              22176           63      352.0
12              18693           93      201.0
89              23128          118      196.0
99              18330           94      195.0
50              17756           92      193.0
38              13680           72      190.0
51              16643           89      187.0
6               22627          121      187.0
11              17480           95      184.0
79              17738           98      181.0
43              19367          107      181.0
90              19758          111      178.0
59              21538          121      178.0
60              16461           93      177.0
82              14691           83      177.0
81              22656          128      177.0
17              17600          100      176.0
88              17776          101      176.0
26              16720           95

In [17]:
#AOV comparison of with and without shop 78 
WO78 = WO607[WO607["shop_id"] != 78]
avgWO78 = WO78["order_amount"].mean()
print("AOV with shop 78: " + "${:,.2f}".format(avgWO607))
print("AOV without shop 78: " + "${:,.2f}".format(avgWO78))



AOV with shop 78: $754.09
AOV without shop 78: $302.58


# Recommendations
After making these financial analyses, I recommend Shopify look deeper into user_id 607's profile, and history, and hold off any actions that this user has to make with this credit card. In addition, reach out to their credit card company and police for further investigation. These actions should be taken to make sure that this is not a credit card fraud.
	
The recommendations below will exclude user_id 607 because of potential credit card fraud. Although finding the AOV is important, Shopify should also consider other metrics such as the most valuable customer and the most valuable store. When it comes to the most valuable customer, user_id 878 has the highest order amount of \\$156,936 compared to the market average of \\$12,042. With this information, the company should focus on keeping this customer since user_id 878 is the one that contributes the most to the order amount.  Another metric that Shopify should adopt is finding the most profitable store. After determining the sales made by each shop, shop_id 78 has the highest sales, \\$2,263,800, while the market average is \\$37,732


In [18]:
#Finding the most valuable customer
pivot3 = WO607.pivot_table(index = "user_id", values = ["order_amount", "total_items"], aggfunc = np.sum)
pivot3 = pivot3.sort_values("order_amount", ascending = False)
print(pivot3.head(10))

         order_amount  total_items
user_id                           
878            156936           24
834            108342           38
787             85707           57
969             84269           50
818             82485           38
775             81114           28
915             80992           29
817             80988           30
855             80915           27
928             80668           28


In [19]:
#Importance of user 878
print("Total Value of user 878: " + "${:,.2f}".format(pivot3.iloc[0,0]))
WO878 = pivot3.drop(878)
avgWO878 = WO878["order_amount"].mean()
print("AOV without user 878: " + "${:,.2f}".format(avgWO878))

Total Value of user 878: $156,936.00
AOV without user 878: $12,042.49


In [20]:
#Finding the most valuable shop
pivot4 = WO607.pivot_table(index = "shop_id", values = ["order_amount"], aggfunc = np.sum)
pivot4 = pivot4.sort_values("order_amount", ascending = False)
print(pivot4.head(10))

         order_amount
shop_id              
78            2263800
89              23128
81              22656
6               22627
42              22176
13              21760
59              21538
71              21320
19              20538
70              20241
